PayloadPkg/OsLoader/OsLoader.c

```diff
@@ -208,11 +208,12 @@ SetupBootImage (
   MultiBoot = &LoadedImage->Image.MultiBoot;
   if (IsElfImage (BootFile->Addr)) {
     DEBUG ((DEBUG_INFO, "Boot image is ELF format...\n"));
-    EntryPoint = LoadElfImage (MultiBoot);
+    EntryPoint = LoadElfImage (BootFile->Addr);
     if (EntryPoint != NULL) {
       if (IsMultiboot (BootFile->Addr)) {
         DEBUG ((DEBUG_INFO, "and Image is Multiboot format\n"));
         SetupMultibootInfo (MultiBoot);
+        UpdateElfMultibootInfo (BootFile->Addr, &MultiBoot->MbInfo);
```

Comment:
Prefer add UpdateElfMultibootInfo() into SetupMultibootInfo() to make this file clean. both parameters in new function could be got from MultiBoot parameter of SetupMultibootInfo().

Reply:
I am OK with that. For all multiple boot support to ELF image, we put it in SetupMultibootInfo(). we could enhance this function later if we really need it.